controllers/argocd/server/clusterrole.go

```diff
@@ -0,0 +1,127 @@
+package server
+
+import (
+	"reflect"
+
+	"github.com/argoproj-labs/argocd-operator/common"
+	"github.com/argoproj-labs/argocd-operator/pkg/mutation"
+	"github.com/argoproj-labs/argocd-operator/pkg/permissions"
+	v1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (sr *ServerReconciler) reconcileClusterRole() error {
+
+	crName := getClusterRoleName(sr.Instance.Name, sr.Instance.Namespace)
+	crLables := common.DefaultResourceLabels(crName, sr.Instance.Name, ServerControllerComponent)
+
+	// ArgoCD instance is not cluster scoped, cleanup any existing cluster role & exit
+	if !sr.ClusterScoped {
+		return sr.deleteClusterRole(crName)
+	}
+
+	sr.Logger.Info("reconciling clusterRole")
+
+	crRequest := permissions.ClusterRoleRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        crName,
+			Labels:      crLables,
+			Annotations: sr.Instance.Annotations,
+		},
```

Comment:
call `argoutil.GetObjMeta` 
I think if we pass an empty string in place of namespace it should make it cluster scoped